test/unit/util/application-logging.test.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const sinon = require('sinon')
+const loggingUtils = require('../../../lib/util/application-logging')
+
+tap.test('truncate', (t) => {
+  t.autoend()
+  t.test('Should truncate string > 1024 chars', (t) => {
+    const longString =
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111' +
+      '1111111111111111111111111111111111111111111111111111111111111111'
+
+    const processedStr = loggingUtils.truncate(longString)
+
+    t.equal(processedStr.length, 1024)
+    t.equal(processedStr.substring(processedStr.length - 3), '...')
+
+    t.end()
+  })
+
+  t.test('Should return non-truncated string when <= 1024 chars', (t) => {
+    const str = 'kenny loggins'
```

Comment:
That's who all of this logging stuff makes you think of, huh?

Reply:
Copied from enricher